### PR TITLE
#changed - foreign key creation when skip-show-database is on

### LINKED ISSUE #875: Cannot create foreign key when skip-show-database is in .cnf
V3.2.0

Cannot create foreign key, references dropdowns are both empty, when they do exist.

Screenshot 2021-02-15 at 10 31 22

### PR DESCRIPTION
**Pull request by gduh**
## Changes:
- Add a warning message when connecting when skip-show-database is ON
- Allows creation of a foreign key using the currently selected database 

## Closes:
- Closes #875 

## Tested:
- Processors:
  - [X] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [ ] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [X] 10.15.x (Catalina)
  - [ ] 11.x (Big Sur)
- Localizations:
  - [X] English
  - [ ] Spanish
  - [ ] Other (please specify)
- Xcode Version: 12.4
  
## Additional notes:
If the skip-show-database option is on, by default, no default database will be automatically selected. 
But, using queries, it's possible to select a database (ex: `use my_database;`). It's also possible to select a default database on the connection panel.

### CODE REVIEW

File: Resources/Localization/es-ES.lproj/Localizable.strings [hunk 1]
```diff
@@ -3233,3 +3233,6 @@
 
 /* New Connection Error informative message */
 "Failed to create new database connection window. Please restart Sequel Ace and try again." = "Failed to create new database connection window. Please restart Sequel Ace and try again.";
+
+/* Warning message during connection in case the variable skip-show-database is set to ON */
```
Comment (Kaspik) [inline Resources/Localization/es-ES.lproj/Localizable.strings:3237]: For future - you need to add it just to english, process will take care of the rest of the languages, but thanks! :)

File: Source/Controllers/MainViewControllers/SPDatabaseDocument.m [hunk 2]
```diff
@@ -417,6 +417,23 @@ - (void)setConnection:(SPMySQLConnection *)theConnection
     // Ensure the connection encoding is set to utf8 for database/table name retrieval
     [mySQLConnection setEncoding:@"utf8mb4"];
 
+    // Check if skip-show-database is set to ON
+    SPMySQLResult *result = [mySQLConnection queryString:@"SHOW VARIABLES LIKE 'skip_show_database'"];
+    [result setReturnDataAsStrings:YES];
+    if(![mySQLConnection queryErrored]) {
```
Comment (Kaspik) [inline Source/Controllers/MainViewControllers/SPDatabaseDocument.m:423]: Can we combine the ifs into one?

File: Source/Controllers/MainViewControllers/SPDatabaseDocument.m [hunk 3]
```diff
@@ -417,6 +417,23 @@ - (void)setConnection:(SPMySQLConnection *)theConnection
...
+        if ([result numberOfRows] == 1) {
+            NSString *skip_show_database = [[result getRowAsDictionary] objectForKey:@"Value"];
+            if ([skip_show_database caseInsensitiveCompare:@"on"] == NSOrderedSame) {
+                NSAlert *alert = [[NSAlert alloc] init];
```
Comment (Kaspik) [inline Source/Controllers/MainViewControllers/SPDatabaseDocument.m:427]: Please use `NSAlert createWarning` or any alternative one that we have in Swift extension :)

**Review by Kaspik — CHANGES_REQUESTED** [review]

File: Source/Controllers/MainViewControllers/SPDatabaseDocument.m (hunk 2 above)
Comment (gduh) [reply Source/Controllers/MainViewControllers/SPDatabaseDocument.m]: It's done. 
`SHOW VARIABLES...` is used more than once to retrieve a single value. I wondered if it was a good idea or not  to create a common method for this. But that was not really the purpose of this change.

File: Source/Controllers/MainViewControllers/SPDatabaseDocument.m (hunk 3 above)
Comment (gduh) [reply Source/Controllers/MainViewControllers/SPDatabaseDocument.m]: It's done.

File: Source/Controllers/MainViewControllers/SPDatabaseDocument.m [hunk 4]
```diff
@@ -420,17 +420,10 @@ - (void)setConnection:(SPMySQLConnection *)theConnection
...
-                NSAlert *alert = [[NSAlert alloc] init];
-                [alert setAlertStyle:NSAlertStyleWarning];
-                [alert setMessageText:NSLocalizedString(@"Warning",@"warning")];
-                [alert setInformativeText:[NSString stringWithFormat:NSLocalizedString(@"The skip-show-database variable of the database server is set to ON. Thus, you won't be able to list databases unless you have the SHOW DATABASES privilege.\n\nHowever, the databases are still accessible directly through SQL queries depending on your privileges.", @"Warning message during connection in case the variable skip-show-database is set to ON")]];
-                [alert addButtonWithTitle:NSLocalizedString(@"OK", @"OK button")];
-                [alert runModal];
-            }
+    if(![mySQLConnection queryErrored] && [result numberOfRows] == 1) {
+        NSString *skip_show_database = [[result getRowAsDictionary] objectForKey:@"Value"];
+        if ([skip_show_database caseInsensitiveCompare:@"on"] == NSOrderedSame) {
```
Comment (Kaspik) [inline Source/Controllers/MainViewControllers/SPDatabaseDocument.m:425]: `isEqualTo:` for NSString comparison :) (sorry I didn't see it previously)

**Review by Kaspik — CHANGES_REQUESTED** [review]
One more thing, and then good to go, thanks!

File: Source/Controllers/MainViewControllers/SPDatabaseDocument.m (hunk 4 above)
Comment (gduh) [reply Source/Controllers/MainViewControllers/SPDatabaseDocument.m:425]: Yes I wondered if I should use `isEqualTo` or not, but with all versions of MySQL / MariaDB, I'm not sur if the value is always `ON` or `On` or `on` or....
So, do you prefer isEqualTo ?

Comment (Kaspik) [reply Source/Controllers/MainViewControllers/SPDatabaseDocument.m:425]: then let's do `skip_show_database.lowercase isEqualTo:@"on"` ?

Comment (gduh) [reply Source/Controllers/MainViewControllers/SPDatabaseDocument.m]: It's done

**Review by Kaspik — APPROVED** [review]
Merge master to resolve conflicts please

**Comment by gduh** [comment]
Master merged

**Review by Kaspik — APPROVED** [review]

**Pull request merged**